models/participants.go

```diff
@@ -0,0 +1,28 @@
+package models
+
+import (
+	"time"
+)
+
+type ParticipantStatus string
+type ParticipantRole string
+
+const (
+	StatusInterested ParticipantStatus = "INTERESTED"
+	StatusNotGoing   ParticipantStatus = "NOT_GOING"
+	StatusCheckedIn  ParticipantStatus = "CHECKED_IN"
+
+	RoleParticipant ParticipantRole = "PARTICIPANT"
+)
+
+type Participant struct {
+	ID        uint64            `json:"-" gorm:"primaryKey"`
+	UserID    uint64            `json:"-"`
+	User      *User             `json:"user,omitempty" gorm:"foreignKey:ID;references:UserID"`
+	EventID   int64             `json:"-"`
+	Event     *Event            `json:"event,omitempty" gorm:"foreignKey:ID;references:EventID"`
```

Comment:
We'll be adding `event` data to all participants which seems a lot of duplicate data, can we remove event data from participant altogether? since event id is required anyways to fetch participants.

Also, `PARTICIPANT` role value of `type ParticipantRole` in a `Participant` table is kind of confusing 😛

Reply:
> We'll be adding event data to all participants which seems a lot of duplicate data

this is at application level only, db will store only event_id (not full event)

> Also, PARTICIPANT role value of type ParticipantRole in a Participant table is kind of confusing 😛

makes sense, will change to ATTENDEE maybe

Reply:
> this is at application level only, db will store only event_id (not full event)

in api response as well, we'll be sending event data for all participants?